jobbergate-agent/jobbergate_agent/utils/compute.py

```diff
@@ -0,0 +1,126 @@
+"""Core module for compute related functions."""
+
+import tracemalloc
+from collections.abc import Callable
+from functools import wraps
+from typing import Any, get_args, cast
+from collections.abc import Iterator
+
+import numpy as np
+from loguru import logger
+from numba import njit
+
+from jobbergate_agent.jobbergate.schemas import InfluxDBMeasure
+
+from jobbergate_agent.jobbergate.constants import INFLUXDB_MEASUREMENT
+
+
+def measure_memory_usage(func: Callable) -> Callable:
+    """Decorator to measure the memory usage of a function.
+
+    Args:
+        func: Function to measure memory usage of.
```

Comment:
```suggestion
        func: Function who's memory usage should be measured.
```

Reply:
done in [77fd610](https://github.com/omnivector-solutions/jobbergate/pull/663/commits/77fd610ad70a8bc1dec2c9395c8c51fd7ba47db4)